src/main/java/appeng/fluids/helper/DualityFluidInterface.java

```diff
@@ -0,0 +1,632 @@
+	private final boolean[] tankChanged;
+	private int isWorking = -1;
+
+	private final MEMonitorPassThrough<IAEItemStack> items = new MEMonitorPassThrough<>( new NullInventory<IAEItemStack>(), AEApi.instance()
+			.storage()
+			.getStorageChannel( IItemStorageChannel.class ) );
+	private final MEMonitorPassThrough<IAEFluidStack> fluids = new MEMonitorPassThrough<>( new NullInventory<IAEFluidStack>(), AEApi.instance()
+			.storage()
+			.getStorageChannel( IFluidStorageChannel.class ) );
+
+	public DualityFluidInterface( final AENetworkProxy networkProxy, final IFluidInterfaceHost ih )
+	{
+		this.gridProxy = networkProxy;
+		this.gridProxy.setFlags( GridFlags.REQUIRE_CHANNEL );
+		this.iHost = ih;
+
+		this.mySource = new MachineSource( this.iHost );
+		this.interfaceRequestSource = new InterfaceRequestSource( this.iHost );
+
+		this.fluids.setChangeSource( this.mySource );
+		this.items.setChangeSource( this.mySource );
+
+		this.tanks = new AEFluidTank[NUMBER_OF_TANKS];
+		this.requireWork = new IAEFluidStack[NUMBER_OF_TANKS];
+		this.configStacks = new IAEFluidStack[NUMBER_OF_TANKS];
+		this.tankChanged = new boolean[NUMBER_OF_TANKS];
+		for( int i = 0; i < NUMBER_OF_TANKS; ++i )
+		{
+			this.tanks[i] = new AEFluidTank( this, TANK_CAPACITY );
+			this.requireWork[i] = null;
+			this.configStacks[i] = null;
+			this.tankChanged[i] = true;
+		}
+		this.storage = new FluidHandlerConcatenate( this.tanks );
+	}
+
+	@Override
+	public <T extends IAEStack<T>> IMEMonitor<T> getInventory( IStorageChannel<T> channel )
+	{
+		if( channel == AEApi.instance().storage().getStorageChannel( IItemStorageChannel.class ) )
+		{
+			if( this.hasConfig() )
+			{
+				return null;
+			}
+
+			return (IMEMonitor<T>) this.items;
+		}
+		else if( channel == AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ) )
+		{
+			if( this.hasConfig() )
+			{
+				return (IMEMonitor<T>) new InterfaceInventory( this );
+			}
+
+			return (IMEMonitor<T>) this.fluids;
+		}
+
+		return null;
+	}
+
+	public IStorageMonitorable getMonitorable( final IActionSource src )
+	{
+		if( Platform.canAccess( this.gridProxy, src ) )
+		{
+			return this;
+		}
+
+		return null;
+	}
+
+	@Override
+	public TickingRequest getTickingRequest( final IGridNode node )
+	{
+		return new TickingRequest( TickRates.Interface.getMin(), TickRates.Interface.getMax(), !this.hasWorkToDo(), true );
+	}
+
+	@Override
+	public TickRateModulation tickingRequest( final IGridNode node, final int ticksSinceLastCall )
+	{
+		if( !this.gridProxy.isActive() )
+		{
+			return TickRateModulation.SLEEP;
+		}
+
+		final boolean couldDoWork = this.updateStorage();
+		return this.hasWorkToDo() ? ( couldDoWork ? TickRateModulation.URGENT : TickRateModulation.SLOWER ) : TickRateModulation.SLEEP;
+	}
+
+	public void notifyNeighbors()
+	{
+		if( this.gridProxy.isActive() )
+		{
+			try
+			{
+				this.gridProxy.getTick().wakeDevice( this.gridProxy.getNode() );
+			}
+			catch( final GridAccessException e )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1166)